sympy/combinatorics/fp_groups.py

```diff
@@ -1003,4 +1005,330 @@ def first_in_class(C, Y=[]):
+        p[psi] = phi
+        p_p[psi] = p_p[lamda]**-1*w**-1*p_p[k]
+        q.append(psi)
+
+def modified_coincidence(C, alpha, beta, w):
+    A_dict = C.A_dict
+    A_dict_inv = C.A_dict_inv
+    table = C.table
+    p_p = C.p_p
+    l = 0
+    q = []
+    modified_merge(C, alpha, beta, w, q)
+    while len(q) > 0:
+        gamma = q.pop(0)
+        for x in A_dict:
+            delta = table[gamma][A_dict[x]]
+            if delta is not None:
+                table[delta][A_dict_inv[x]] = None
+                mu = C.rep(gamma)
+                nu = C.rep(delta)
+                if table[mu][A_dict[x]] is not None:
+                    v = p_p[delta]**-1*C.P[gamma][A_dict[x]]**-1*p_p[gamma]*C.P[mu][A_dict[x]]
+                    modified_merge(C, nu, table[mu][A_dict[x]], v, q)
+                elif table[nu][A_dict_inv[x]] is not None:
+                    v = p_p[gamma]**-1*C.P[gamma][A_dict[x]]*p_p[delta]*C.P[mu][A_dict_inv[x]]
+                else:
+                    table[mu][A_dict[x]] = nu; table[nu][A_dict_inv[x]] = mu
+                    v = p_p[gamma]**-1*C.P[gamma][A_dict[x]]*p_p[delta]
+                    C.P[mu][A_dict[x]] = v
+                    C.P[nu][A_dict_inv[x]] = v**-1
+
+
+def elimination_technique_1(C):
+    rels = list(C.reidemeister_relators)
+    # the shorter relators are examined first so that generators selected for
+    # elimination will have shorter strings as equivalent
+    rels.sort(reverse=True)
+    gens = list(C.schreier_generators)
+    # TODO: redundant generator can also be present as inverse in relator
+    redundant_gens = {}
+    contained_gens = []
+    # examine each relator in relator list for any generator occuring exactly
+    # once
+    next_i = False
+    for i in range(len(rels) -1, -1, -1):
+        rel = rels[i]
+        for gen in redundant_gens:
+            if any([gen.array_form[0][0] == r[0] for r in rel.array_form]):
+                next_i = True
+                break
+        if next_i:
+            next_i = False
+            continue
+        j = len(gens) - 1
+        while j >= 0:
+            gen = gens[j]
+            if rel.generator_exponent_sum(gen) == 1 and gen not in contained_gens:
+                k = rel.exponent_sum_word(gen)
+                gen_index = rel.index(gen**k)
+                bk = rel.subword(gen_index + 1, len(rel))
+                fw = rel.subword(0, gen_index)
+                redundant_gens[gen] = (bk*fw)**(-1*k)
+                contained_gens.extend((bk*fw).contains_generators())
+                del rels[i]; del gens[j]
+                break
+            j -= 1
+    # eliminate the redundant generator from remaing relators
+    for i, gen in product(range(len(rels)), redundant_gens):
+        rels[i] = rels[i].eliminate_word(gen, redundant_gens[gen])
+    rels.sort()
+    C.reidemeister_relators = rels
+    C.schreier_generators = gens
+
+def elimination_technique_2(C):
+    pass
+
+def simplification_technique_1(C):
```

Comment:
Will probably see a few things for this, in the morning. It wouldn't be hard to implement this technique.